tiledb/array_schema.py

```diff
@@ -0,0 +1,442 @@
+        """Create an ArraySchema for a Filestore Array from a given file.
+        If a uri is not given, then create a default schema."""
+        _ctx = ctx or default_ctx()
+
+        return ArraySchema(_lt_obj=lt.Filestore._schema_create(_ctx, uri))
+
+    def __eq__(self, other):
+        """Instance is equal to another ArraySchema"""
+        if not isinstance(other, ArraySchema):
+            return False
+        nattr = self.nattr
+        if nattr != other.nattr:
+            return False
+        if (
+            self.sparse != other.sparse
+            or self.cell_order != other.cell_order
+            or self.tile_order != other.tile_order
+        ):
+            return False
+        if self.capacity != other.capacity:
+            return False
+        if self.domain != other.domain:
+            return False
+        if self.coords_filters != other.coords_filters:
+            return False
+        for i in range(nattr):
+            if self.attr(i) != other.attr(i):
+                return False
+        return True
+
+    def __len__(self):
+        """Returns the number of Attributes in the ArraySchema"""
+        return self._nattr
+
+    def __iter__(self):
+        """Returns a generator object that iterates over the ArraySchema's Attribute objects"""
+        return (self.attr(i) for i in range(self.nattr))
+
+    @property
+    def ctx(self) -> "Ctx":
+        """The array schema's context
+
+        :rtype: tiledb.Ctx
+        """
+        return self._ctx
+
+    def check(self) -> bool:
+        """Checks the correctness of the array schema
+
+        :rtype: None
+        :raises: :py:exc:`tiledb.TileDBError` if invalid
+        """
+        return self._check()
+
+    @property
+    def sparse(self) -> bool:
+        """True if the array is a sparse array representation
+
+        :rtype: bool
+        :raises: :py:exc:`tiledb.TileDBError`
+
+        """
+        return self._array_type == lt.ArrayType.SPARSE
+
+    @property
+    def allows_duplicates(self) -> bool:
+        """Returns True if the (sparse) array allows duplicates."""
+
+        if not self.sparse:
+            raise lt.TileDBError(
+                "ArraySchema.allows_duplicates does not apply to dense arrays"
+            )
+
+        return self._allows_dups
+
+    @property
+    def capacity(self) -> int:
+        """The array capacity
+
+        :rtype: int
+        :raises: :py:exc:`tiledb.TileDBError`
+
+        """
+        return self._capacity
+
+    @property
+    def cell_order(self) -> str:
+        """The cell order layout of the array.
+
+        :rtype: str
+        """
+        return tiledb_layout_string(self._cell_order)
+
+    @property
+    def tile_order(self) -> str:
+        """The tile order layout of the array.
+
+        :rtype: str
+        :raises: :py:exc:`tiledb.TileDBError`
+
+        """
+        layout_string = tiledb_layout_string(self._cell_order)
+        if self.cell_order == "hilbert":
```

Comment:
- `self._cell_order` should almost certainly be `self._tile_order` 
- is the comparison correct as is or should it be `if layout_string == "hilbert"`?
cc @nguyenv @ihnorton